spec/relativetimeformat.html

```diff
@@ -1,61 +1,261 @@
+        1. Perform ! CreateDataPropertyOrThrow(_nfOptions_, `"minimumIntegerDigits"`, *2*).
+        1. Let _relativeTimeFormat_.[[NumberFormat]] be ? Construct(%NumberFormat%, « _nfLocale_, _nfOptions_ »).
+        1. Let _prLocale_ be CreateArrayFromList(« _locale_ »).
+        1. Let _prOptions_ be ObjectCreate(%ObjectPrototype%).
+        1. Perform ! CreateDataPropertyOrThrow(_prOptions_, *"style"*, *"cardinal"*).
+        1. Let _relativeTimeFormat_.[[PluralRules]] be ? Construct(%PluralRules%, « _prLocale_, _prOptions_ »).
+        1. Set _relativeTimeFormat_.[[InitializedRelativeTimeFormat]] to *true*.
         1. Return _relativeTimeFormat_.
       </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-PartitionRelativeTimePattern" aoid="PartitionRelativeTimePattern">
+      <h1>PartitionRelativeTimePattern (relativeTimeFormat, x)</h1>
+
+      <p>
+        When the FormatRelativeTime abstract operation is called with arguments _relativeTimeFormat_  and _value_, it returns a String value representing _value_ (interpreted as a time value as specified in ES2016, 20.3.1.1) according to the effective locale and the formatting options of _relativeTimeFormat_.
+      </p>
 
+      <emu-alg>
+        1. Assert: _relativeTimeFormat_.[[InitializedRelativeTimeFormat]] is *true*.
+        1. Assert: Type(_value_) is a Number.
+        1. If isFinite(_x_) is *false*, then throw a *RangeError* exception.
+        1. Let _fields_ be _relativeTimeFormat_.[[Fields]].
+        1. Let _style_ be _relativeTimeFormat_.[[Style]].
+        1. Let _now_ be Call(%Date_now%, *undefined*).
+        1. Let _ms_ be _x_ - _now_.
+        1. Let _units_ be ComputeTimeUnits(_ms_).
+        1. If _relativeTimeFormat_.[[Unit]] is equal to "best fit", then
+          1. Let _unit_ be GetBestMatchUnit(_units_).
+        1. Else
+          1. Let _unit_ be _relativeTimeFormat_.[[unit]].
+        1. If _style_ is equal to "short", then
+          1. Let _entry_ be _unit_ + "-short".
+        1. Else if _style_ is equal to "narrow", then
+          1. Let _entry_ be _unit_ + "-narrow".
+        1. Else
+          1. Let _entry_ be _unit_.
+        1. Let _exists_ be ? HasProperty(_fields_, _entry_).
+        1. If _exists_ is *false*, then
+          1. Let _entry_ be _unit_.
+        1. Let _patterns_ be ? Get(_fields_, _entry_).
+        1. Let _v_ be _units_.[[&lt;_unit_&gt;]].
+        1. If _v_ is between -2 (inclusive) and 2 (inclusive), then
+          1. Let _exists_ be ? HasProperty(_patterns_, ToString(_v_)).
+          1. If _exists_ is *true*, then
+            1. Let _result_ be Get(_patterns_, ToString(_v_)).
+            1. Return _result_.
+        1. If _v_ is less than 0, then
+          1. Let _tl_ be *"past"*.
+        1. Else
+          1. Let _tl_ be *"future"*.
+        1. Let _po_ be ? Get(_patterns_, _tl_).
+        1. Let _fv_ be ! FormatNumber(_relativeTimeFormat_.[[NumberFormat]], _v_).
```

Comment:
Sure!

We decided we need the same formatting options in PluralRules as we have in NumberFormat because only after we format the number we know if it has fraction part of not, which affects plural rule selected - that in turn, means that we can abstract some of the NumberFormat work and just reuse it instead of duplicating - https://github.com/tc39/proposal-intl-plural-rules/pull/13

I suspect that we may want to do something similar here.

On top of that, I started discussion about potential simplification of the 5 formatting params - https://github.com/tc39/ecma402/issues/91 .